use the system malloc so we can easily
switch to others, like libumem. I'm not
entirely sure this should be changed upstream
yet.

--- tcsh-6.24.07/config_f.h.orig	2023-01-02 16:10:49.993654296 +0100
+++ tcsh-6.24.07/config_f.h	2023-01-02 16:11:29.502405676 +0100
@@ -147,7 +147,7 @@
  *		This can be much slower and no memory statistics will be
  *		provided.
  */
-#if defined(__MACHTEN__) || defined(PURIFY) || defined(MALLOC_TRACE) || defined(_OSD_POSIX) || defined(__MVS__) || defined (__CYGWIN__) || defined(__GLIBC__) || defined(__OpenBSD__) || defined(__APPLE__) || defined (__ANDROID__) || defined(__NetBSD__) || !defined(HAVE_WORKING_SBRK)
+#if defined(__MACHTEN__) || defined(PURIFY) || defined(MALLOC_TRACE) || defined(_OSD_POSIX) || defined(__MVS__) || defined (__CYGWIN__) || defined(__GLIBC__) || defined(__OpenBSD__) || defined(__APPLE__) || defined (__ANDROID__) || defined(__NetBSD__) || !defined(HAVE_WORKING_SBRK) || defined(SOLARIS2)
 # define SYSMALLOC
 #else
 # undef SYSMALLOC
